merk/src/proofs/query/mod.rs

```diff
@@ -169,18 +207,24 @@ pub enum QueryItem {
     RangeFrom(RangeFrom<Vec<u8>>),
     RangeTo(RangeTo<Vec<u8>>),
     RangeToInclusive(RangeToInclusive<Vec<u8>>),
+    RangeAfter(RangeFrom<Vec<u8>>),
+    RangeAfterTo(Range<Vec<u8>>),
+    RangeAfterToInclusive(RangeInclusive<Vec<u8>>),
 }
 
 impl QueryItem {
-    pub fn lower_bound(&self) -> &[u8] {
+    pub fn lower_bound(&self) -> (&[u8], bool) {
```

Comment:
The bool here seems to signify not_present, shouldn't we have it be present? like the upper bound?

Reply:
nevermind, i understand now